zcash_transparent/src/coinbase.rs

```diff
@@ -0,0 +1,110 @@
+//! Types, constants, and functions related to coinbase transactions.
+
+use alloc::{fmt, vec::Vec};
+
+/// The maximum length of the coinbase script sig field.
+///
+/// # Consensus
+///
+/// > A coinbase transaction script MUST have length in {2 .. 100} bytes.
+///
+/// <https://zips.z.cash/protocol/protocol.pdf#txnconsensus>
+pub const MAX_COINBASE_SCRIPT_LEN: usize = 100;
+
+/// The minimum length of the coinbase script sig field.
+///
+/// # Consensus
+///
+/// > A coinbase transaction script MUST have length in {2 .. 100} bytes.
+///
+/// <https://zips.z.cash/protocol/protocol.pdf#txnconsensus>
+pub const MIN_COINBASE_SCRIPT_LEN: usize = 2;
+
+/// The maximum length of the encoded height in the coinbase script sig field.
+///
+/// # Consensus
+///
+/// > The length of `heightBytes` MUST be in the range {1 .. 5}. Then the encoding is the length of
+/// > `heightBytes` encoded as one byte, followed by heightBytes itself.
+///
+/// <https://zips.z.cash/protocol/protocol.pdf#txnconsensus>
+pub const MAX_COINBASE_HEIGHT_LEN: usize = 6;
+
+/// The maximum length of the miner data in the coinbase script sig field.
+pub const MAX_MINER_DATA_LEN: usize = MAX_COINBASE_SCRIPT_LEN - MAX_COINBASE_HEIGHT_LEN;
```

Comment:
This seems fine for constructing new coinbase transactions, but its use in Zebra's `parse_coinbase_height()` function seems wrong if coinbase transactions with heights that are encoded to fewer bytes are supposed to allow more miner data.

Reply:
I think it'll be best to keep the limit static. The bitcoin developer website also says: _"Arbitrary data not exceeding 100 bytes minus the (4) height bytes."_